ee/modules/030-cloud-provider-openstack/hooks/discover_volume_types.go

```diff
@@ -49,7 +49,7 @@ func handleDiscoverVolumeTypes(input *go_hook.HookInput) error {
 	storageClassesMap := make(map[string]string, len(openstackVolumeTypes))
 
 	for _, vt := range openstackVolumeTypes {
-		storageClassesMap[strings.ToLower(strings.ReplaceAll(vt, "_", ""))] = vt
+		storageClassesMap[strings.ToLower(strings.ReplaceAll(strings.ReplaceAll(vt, "_", ""), " ", "-"))] = vt
```

Comment:
This is an ok solution, but I'd like to fix it once and forever.
There are functions that converts strings to valid DNS1123 domain labels, e.g., 

```go
const (
	DNS1123NameMaximumLength         = 63
	DNS1123NotAllowedCharacters      = "[^-a-z0-9]"
	DNS1123NotAllowedStartCharacters = "^[^a-z0-9]+"
)

func makeDNS1123Compatible(name string) string {
	name = strings.ToLower(name)

	nameNotAllowedChars := regexp.MustCompile(DNS1123NotAllowedCharacters)
	name = nameNotAllowedChars.ReplaceAllString(name, "")

	nameNotAllowedStartChars := regexp.MustCompile(DNS1123NotAllowedStartCharacters)
	name = nameNotAllowedStartChars.ReplaceAllString(name, "")

	if len(name) > DNS1123NameMaximumLength {
		name = name[0:DNS1123NameMaximumLength]
	}

	return name
}

```

What do you think?

Reply:
Good point. I had a thought about making a generic solution. Then I thought that we want to change spaces to dashes because it gives more proper names for StorageClasses. But as I think now, it's better just to remove all invalid characters as it was done for underscores previously. I will apply your suggestion.

Reply:
Done